app/services/pii/cacher.rb

```diff
@@ -0,0 +1,31 @@
+module Pii
+  class Cacher
+    def initialize(user, user_session)
+      @user = user
+      @user_session = user_session
+    end
+
+    def save(password, profile = user.active_profile)
+      return unless profile
+      decrypted_pii = profile.decrypt_pii(password)
+      pii_json = decrypted_pii.to_json
+      encrypted_pii = encryptor.encrypt(pii_json)
+      user_session[:encrypted_pii] = encrypted_pii
+    end
```

Comment:
I don't fully understand the purpose of this Cacher class. It seems like in order for the encrypted PII to be cached, the user must have an active profile. But if they already have an active profile, that means they already have encrypted PII in the DB, right? If so, why can't the encrypted PII be fetched directly from the DB without the session as a middleman? 

From what I can tell, here are the sequence of events:

1. User with an active profile signs in
2. Their encrypted PII is fetched from the DB and stored in the session (after_action in users/sessions_controller)
3. Since their profile is already active, meaning they've already proofed, their PII is not going to be modified in this session
4. The encrypted PII is then fetched from the session and decrypted in 2 places: 1) the ProfileController in order to display to the user their verified attributes, 2) in SamlIdpController to assert attributes back to the SP

If this is correct, it looks like the caching is not necessary since the data is already available in the DB.